### ISSUE TITLE
E2E tests not working with minikube

### ISSUE THREAD

**Issue by mukundansundar** [dapr/dapr#2102]
## In what area(s)?

> /area runtime
> /area operator
> /area placement
> /area docs

/area test-and-release

## What version of Dapr?

Latest master version of Dapr as of 9/24/2020

## Expected Behavior

```
2020/09/24 14:54:41 Running setup...
2020/09/24 14:54:41 Installing test apps...
2020/09/24 14:54:41 Adding app {stateapp 0  true e2e-stateapp:dev docker.io/<repo> 1 true   4.0 0.5 512Mi 250Mi}
2020/09/24 14:54:57 Running tests...
2020/09/24 14:54:57 Waiting until service ingress is ready for stateapp...
2020/09/24 14:55:02 Service ingress for stateapp is ready...
PASS
2020/09/24 14:55:37 Running teardown...

```
## Actual Behavior

The test times out.

The daprd container does not start and the pod goes into a CrashLoopBackoff state. 

```
stateapp-58dfb67cf5-xm2bt                1/2     CrashLoopBackOff    3          74s
```

## Steps to Reproduce the Problem

1. Get the latest master branch of dapr/dapr.
2. Export the following variables 
```
export DAPR_REGISTRY=docker.io/your_dockerhub_id
export DAPR_TAG=dev
export DAPR_NAMESPACE=dapr-tests
export DAPR_TEST_ENV=minikube
export DAPR_TEST_MINIKUBE_IP=`minikube ip`
export DAPR_TEST_REGISTRY=docker.io/your_dockerhub_id
export DARP_TEST_TAG=dev
```
3. Run the following commands from the running E2E step by step guide 
https://github.com/dapr/dapr/blob/master/tests/docs/running-e2e-test.md#option-2-step-by-step-guide
```
make create-test-namespace
make setup-helm-init
make setup-test-env-redis
# Build Linux binaries
make build-linux

# Build Docker image with Linux binaries
make docker-build

# Push docker image to your dockerhub registry
make docker-push

# Deploy Dapr runtime to your cluster
make docker-deploy-k8s

make setup-test-components


make build-e2e-app-stateapp


make push-e2e-app-stateapp
```
4. Running the code 
From VS Code, remove the build tag from the top of the file 
<dapr repo>/tests/e2e/stateapp/stateapp_test.go
```
// +build e2e 
```
remove the above line. 

Run the TestStateTransactionApps test  code from VSCode.

The logs from the stateapp container is as follows 

```
time="2020-09-24T21:50:06.902106074Z" level=info msg="starting Dapr Runtime -- version edge -- commit v0.10.0-112-g9a4788d-dirty" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.902164955Z" level=info msg="log level set to: info" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.902379863Z" level=info msg="metrics server started on :9090/" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.metrics type=log ver=edge
time="2020-09-24T21:50:06.902572442Z" level=info msg="loading default configuration" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.902720857Z" level=info msg="kubernetes mode configured" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.902778828Z" level=info msg="app id: stateapp" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.902799661Z" level=info msg="mTLS enabled. creating sidecar authenticator" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.902945395Z" level=info msg="trust anchors and cert chain extracted successfully" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime.security type=log ver=edge
time="2020-09-24T21:50:06.902958811Z" level=info msg="authenticator created" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.910047763Z" level=info msg="application protocol: http. waiting on port 3000.  This will block until the app is listening on that port." app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.913368914Z" level=info msg="application discovered on port 3000" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.91379916Z" level=info msg="application configuration loaded" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.913855823Z" level=info msg="Initialized name resolution to kubernetes" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.914884244Z" level=info msg="component loaded. name: kubernetes, type: secretstores.kubernetes" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.923852583Z" level=info msg="component loaded. name: statestore, type: state.redis" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.926727736Z" level=info msg="component loaded. name: messagebus, type: pubsub.redis" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
time="2020-09-24T21:50:06.926943418Z" level=info msg="waiting for all outstanding components to be processed" app_id=stateapp instance=stateapp-58dfb67cf5-xm2bt scope=dapr.runtime type=log ver=edge
```

after this the daprd container is restarted again and again because the liveness probe fails. 

```
Normal   Pulled     21s               kubelet            Successfully pulled image "docker.io/<repo>/daprd:dev-linux-amd64" in 902.3689ms
  Normal   Pulling    5s (x2 over 22s)  kubelet            Pulling image "docker.io/<repo>/daprd:dev-linux-amd64"
  Warning  Unhealthy  5s (x3 over 17s)  kubelet            Liveness probe failed: Get "http://172.17.0.9:3500/v1.0/healthz": dial tcp 172.17.0.9:3500: connect: connection refused
  Normal   Killing    5s                kubelet            Container daprd failed liveness probe, will be restarted
```

**Comment by skyao** [comment]
@mukundansundar  I submmit two similar issues, you can have a look to see if there are some relationship.

#2109 
#2108

**Comment by artursouza** [comment]
They did work for me but I will create a PR fix the tricks. Also, using the uber make target did not work because it was failing to push images, it only worked after I ran one step at a time.

```
$ make test-e2e-all
# Note: we can set -p 2 to run two tests apps at a time, because today we do not share state between
# tests. In the future, if we add any tests that modify global state (such as dapr config), we'll 
# have to be sure and run them after the main test suite, so as not to alter the state of a running
# test. Minikube runs with -p 1 due to limitation in local resources.
GOOS=darwin DAPR_TEST_NAMESPACE=dapr-tests DAPR_TEST_TAG=dev-linux-amd64 DAPR_TEST_REGISTRY=artursouza DAPR_TEST_MINIKUBE_IP=192.168.64.5 go test -p 1 -count=1 -v -tags=e2e ./tests/e2e/...
2020/09/28 19:45:49 Running setup...
2020/09/28 19:45:49 Installing test apps...
2020/09/28 19:45:49 Adding app {actorapp 0  true e2e-actorapp:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:46:32 Running tests...
=== RUN   TestActorActivation
2020/09/28 19:46:32 Waiting until service ingress is ready for actorapp...
2020/09/28 19:46:32 Service ingress for actorapp is ready...
=== RUN   TestActorActivation/Actor_deactivates_due_to_timeout.
getting logs, the current time is 2020-09-28 19:46:33.343715 -0700 PDT m=+44.288509151
getting logs, the current time is 2020-09-28 19:46:53.344664 -0700 PDT m=+64.289235630
=== RUN   TestActorActivation/Actor_does_not_deactivate_since_there_is_no_timeout.
--- PASS: TestActorActivation (22.13s)
    --- PASS: TestActorActivation/Actor_deactivates_due_to_timeout. (20.00s)
    --- PASS: TestActorActivation/Actor_does_not_deactivate_since_there_is_no_timeout. (1.01s)
PASS
2020/09/28 19:46:54 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/actor_activation	66.170s
2020/09/28 19:46:56 Running setup...
2020/09/28 19:46:56 Installing test apps...
2020/09/28 19:46:56 Adding app {actorfeatures 0  true e2e-actorfeatures:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:47:06 Running tests...
=== RUN   TestActorFeatures
2020/09/28 19:47:06 Waiting until service ingress is ready for actorfeatures...
2020/09/28 19:47:06 Service ingress for actorfeatures is ready...
=== RUN   TestActorFeatures/Actor_state.
=== RUN   TestActorFeatures/Actor_reminder.
=== RUN   TestActorFeatures/Actor_timer.
=== RUN   TestActorFeatures/Actor_concurrency_same_actor_id.
=== RUN   TestActorFeatures/Actor_concurrency_different_actor_ids.
=== RUN   TestActorFeatures/Actor_fails_over_to_another_hostname.
=== RUN   TestActorFeatures/Actor_rebalance_to_another_hostname.
=== RUN   TestActorFeatures/Get_actor_metadata
--- PASS: TestActorFeatures (139.39s)
    --- PASS: TestActorFeatures/Actor_state. (20.04s)
    --- PASS: TestActorFeatures/Actor_reminder. (20.01s)
    --- PASS: TestActorFeatures/Actor_timer. (25.01s)
    --- PASS: TestActorFeatures/Actor_concurrency_same_actor_id. (10.01s)
    --- PASS: TestActorFeatures/Actor_concurrency_different_actor_ids. (5.01s)
    --- PASS: TestActorFeatures/Actor_fails_over_to_another_hostname. (9.07s)
    --- PASS: TestActorFeatures/Actor_rebalance_to_another_hostname. (38.07s)
    --- PASS: TestActorFeatures/Get_actor_metadata (11.04s)
PASS
2020/09/28 19:49:25 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/actor_features	149.026s
2020/09/28 19:49:29 Running setup...
2020/09/28 19:49:29 Installing test apps...
2020/09/28 19:49:29 Adding app {allowlists-caller 0  true e2e-service_invocation:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:49:29 Adding app {allowlists-callee-http 0  true e2e-service_invocation:dev-linux-amd64 artursouza 1 false  allowlistsappconfig 4.0 0.5 512Mi 250Mi}
2020/09/28 19:49:29 Adding app {allowlists-callee-grpc 0 grpc true e2e-service_invocation_grpc:dev-linux-amd64 artursouza 1 false  allowlistsgrpcappconfig 4.0 0.5 512Mi 250Mi}
2020/09/28 19:50:03 Running tests...
=== RUN   TestServiceInvocationWithAllowLists
2020/09/28 19:50:03 Waiting until service ingress is ready for allowlists-caller...
2020/09/28 19:50:03 Service ingress for allowlists-caller is ready...
    allowlists_service_invocation_test.go:121: externalURL is '192.168.64.5:32603'
=== RUN   TestServiceInvocationWithAllowLists/Test_allow_with_callee_side_http
    allowlists_service_invocation_test.go:133: checking err...
    allowlists_service_invocation_test.go:137: unmarshalling..{"message":"opAllow is called"}
        
=== RUN   TestServiceInvocationWithAllowLists/Test_deny_with_callee_side_http
    allowlists_service_invocation_test.go:133: checking err...
    allowlists_service_invocation_test.go:137: unmarshalling..{"message":"rpc error: code = PermissionDenied desc = access control policy has denied access to appid: allowlists-caller operation: opDeny verb: POST"}
        
=== RUN   TestServiceInvocationWithAllowLists/Test_allow_with_callee_side_grpc
    allowlists_service_invocation_test.go:154: url is 'http://192.168.64.5:32603/grpctogrpctest'
    allowlists_service_invocation_test.go:159: checking err...
    allowlists_service_invocation_test.go:163: unmarshalling..{"message":"success"}
        
--- PASS: TestServiceInvocationWithAllowLists (1.04s)
    --- PASS: TestServiceInvocationWithAllowLists/Test_allow_with_callee_side_http (0.02s)
    --- PASS: TestServiceInvocationWithAllowLists/Test_deny_with_callee_side_http (1.00s)
    --- PASS: TestServiceInvocationWithAllowLists/Test_allow_with_callee_side_grpc (0.02s)
PASS
2020/09/28 19:50:04 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/allowlists	35.825s
2020/09/28 19:50:06 Running setup...
2020/09/28 19:50:06 Installing test apps...
2020/09/28 19:50:06 Adding app {bindinginput 0  true e2e-binding_input:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:50:06 Adding app {bindingoutput 0  true e2e-binding_output:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:50:27 Running tests...
=== RUN   TestBindings
2020/09/28 19:50:27 Waiting until service ingress is ready for bindingoutput...
2020/09/28 19:50:28 Service ingress for bindingoutput is ready...
2020/09/28 19:50:28 Waiting until service ingress is ready for bindinginput...
2020/09/28 19:50:28 Service ingress for bindinginput is ready...
--- PASS: TestBindings (6.24s)
PASS
2020/09/28 19:50:34 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/bindings	28.170s
2020/09/28 19:50:37 Running setup...
2020/09/28 19:50:37 Installing test apps...
2020/09/28 19:50:37 Adding app {hellobluedapr 0  true e2e-hellodapr:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:50:37 Adding app {hellogreendapr 0  true e2e-hellodapr:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:50:37 Adding app {helloenvtestdapr 0  true e2e-hellodapr:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:51:10 Running tests...
=== RUN   TestHelloDapr
=== RUN   TestHelloDapr/green_dapr
2020/09/28 19:51:10 Waiting until service ingress is ready for hellogreendapr...
2020/09/28 19:51:10 Service ingress for hellogreendapr is ready...
=== RUN   TestHelloDapr/blue_dapr
2020/09/28 19:51:10 Waiting until service ingress is ready for hellobluedapr...
2020/09/28 19:51:10 Service ingress for hellobluedapr is ready...
=== RUN   TestHelloDapr/envTest_dapr
2020/09/28 19:51:10 Waiting until service ingress is ready for helloenvtestdapr...
2020/09/28 19:51:10 Service ingress for helloenvtestdapr is ready...
--- PASS: TestHelloDapr (1.17s)
    --- PASS: TestHelloDapr/green_dapr (0.01s)
    --- PASS: TestHelloDapr/blue_dapr (0.01s)
    --- PASS: TestHelloDapr/envTest_dapr (1.16s)
=== RUN   TestScaleReplicas
--- PASS: TestScaleReplicas (13.02s)
=== RUN   TestScaleAndRestartInstances
--- PASS: TestScaleAndRestartInstances (29.05s)
PASS
2020/09/28 19:51:53 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/hellodapr	77.233s
2020/09/28 19:52:00 Running setup...
2020/09/28 19:52:00 Installing test apps...
2020/09/28 19:52:00 Adding app {httpmetrics 0  true e2e-hellodapr:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:52:00 Adding app {grpcmetrics 0  true e2e-stateapp:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:52:23 Running tests...
=== RUN   TestMetrics
Forwarding from 127.0.0.1:55733 -> 3500
Forwarding from [::1]:55733 -> 3500
Forwarding from 127.0.0.1:55734 -> 9090
Forwarding from [::1]:55734 -> 9090
=== RUN   TestMetrics/http_metrics
Handling connection for 55733
Handling connection for 55734
Forwarding from 127.0.0.1:55738 -> 50001
Forwarding from [::1]:55738 -> 50001
Forwarding from 127.0.0.1:55739 -> 9090
Forwarding from [::1]:55739 -> 9090
=== RUN   TestMetrics/grpc_metrics
Handling connection for 55738
Handling connection for 55739
--- PASS: TestMetrics (0.62s)
    --- PASS: TestMetrics/http_metrics (0.06s)
    --- PASS: TestMetrics/grpc_metrics (0.37s)
PASS
2020/09/28 19:52:23 Running teardown...
2020/09/28 19:52:24 Closed port fowarding
ok  	github.com/dapr/dapr/tests/e2e/metrics	24.649s
Enter TestMain
2020/09/28 19:52:27 Creating TestRunner
2020/09/28 19:52:27 Starting TestRunner
2020/09/28 19:52:27 Running setup...
2020/09/28 19:52:27 Installing test apps...
2020/09/28 19:52:27 Adding app {pubsub-publisher 0  true e2e-pubsub-publisher:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:52:27 Adding app {pubsub-subscriber 0  true e2e-pubsub-subscriber:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:52:48 Running tests...
=== RUN   TestPubSub
    pubsub_test.go:191: Enter TestPubSub
2020/09/28 19:52:48 Waiting until service ingress is ready for pubsub-publisher...
2020/09/28 19:52:48 Service ingress for pubsub-publisher is ready...
2020/09/28 19:52:48 Waiting until service ingress is ready for pubsub-subscriber...
2020/09/28 19:52:48 Service ingress for pubsub-subscriber is ready...
=== RUN   TestPubSub/pubsubtest1
2020/09/28 19:52:50 Sending first publish app at url http://192.168.64.5:31791/tests/publish and body '{"topic":"pubsub-a-topic","data":"message-23"}', this log will not print for subsequent messages for same topic
2020/09/28 19:52:50 Sending first publish app at url http://192.168.64.5:31791/tests/publish and body '{"topic":"pubsub-b-topic","data":"message-78"}', this log will not print for subsequent messages for same topic
2020/09/28 19:52:51 Sending first publish app at url http://192.168.64.5:31791/tests/publish and body '{"topic":"pubsub-c-topic","data":"message-20"}', this log will not print for subsequent messages for same topic
2020/09/28 19:52:56 Publishing using url http://192.168.64.5:32106/tests/get
2020/09/28 19:52:56 subscriber receieved 100 messages on pubsub-a-topic, 100 on pubsub-b-topic and 100 on pubsub-c-topic
--- PASS: TestPubSub (8.43s)
    --- PASS: TestPubSub/pubsubtest1 (6.31s)
PASS
2020/09/28 19:52:56 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/pubsub	29.225s
Enter TestMain
2020/09/28 19:52:59 Creating TestRunner
2020/09/28 19:52:59 Starting TestRunner
2020/09/28 19:52:59 Running setup...
2020/09/28 19:52:59 Installing components...
2020/09/28 19:52:59 Installing init apps...
2020/09/28 19:52:59 Adding app {runtime-init -1  true e2e-runtime_init:dev-linux-amd64 artursouza 1 false   4.0 0.5 512Mi 250Mi}
2020/09/28 19:53:13 Installing test apps...
2020/09/28 19:53:13 Adding app {runtime 0  true e2e-runtime:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:53:24 Running tests...
=== RUN   TestRuntimeInitPubsub
    runtime_test.go:114: Enter TestRuntimeInitPubsub
2020/09/28 19:53:24 Waiting until service ingress is ready for runtime...
2020/09/28 19:53:24 Service ingress for runtime is ready...
--- PASS: TestRuntimeInitPubsub (0.11s)
=== RUN   TestRuntimeInitBindings
    runtime_test.go:137: Enter TestRuntimeInitBindings
2020/09/28 19:53:24 Waiting until service ingress is ready for runtime...
2020/09/28 19:53:24 Service ingress for runtime is ready...
--- PASS: TestRuntimeInitBindings (0.00s)
PASS
2020/09/28 19:53:24 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/runtime	25.774s
2020/09/28 19:53:27 Running setup...
2020/09/28 19:53:27 Installing test apps...
2020/09/28 19:53:27 Adding app {secretapp 0  true e2e-secretapp:dev-linux-amd64 artursouza 1 true  secretappconfig 4.0 0.5 512Mi 250Mi}
2020/09/28 19:53:41 Running tests...
=== RUN   TestSecretApp
2020/09/28 19:53:41 Waiting until service ingress is ready for secretapp...
2020/09/28 19:53:41 Service ingress for secretapp is ready...
=== RUN   TestSecretApp/Test_get_with_empty_request_response_for_single_app_and_single_hop
=== RUN   TestSecretApp/Test_get_a_single_item_for_single_app_and_single_hop
=== RUN   TestSecretApp/Test_unallowed_secret
--- PASS: TestSecretApp (1.15s)
    --- PASS: TestSecretApp/Test_get_with_empty_request_response_for_single_app_and_single_hop (0.00s)
    --- PASS: TestSecretApp/Test_get_a_single_item_for_single_app_and_single_hop (0.01s)
    --- PASS: TestSecretApp/Test_unallowed_secret (0.00s)
PASS
2020/09/28 19:53:42 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/secretapp	14.686s
2020/09/28 19:53:45 Running setup...
2020/09/28 19:53:45 Installing test apps...
2020/09/28 19:53:45 Adding app {serviceinvocation-caller 0  true e2e-service_invocation:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:53:45 Adding app {serviceinvocation-callee-0 0  true e2e-service_invocation:dev-linux-amd64 artursouza 1 false   4.0 0.5 512Mi 250Mi}
2020/09/28 19:53:45 Adding app {serviceinvocation-callee-1 0  true e2e-service_invocation:dev-linux-amd64 artursouza 1 false   4.0 0.5 512Mi 250Mi}
2020/09/28 19:53:45 Adding app {grpcapp 0 grpc true e2e-service_invocation_grpc:dev-linux-amd64 artursouza 1 false   4.0 0.5 512Mi 250Mi}
2020/09/28 19:54:24 Running tests...
=== RUN   TestServiceInvocation
2020/09/28 19:54:24 Waiting until service ingress is ready for serviceinvocation-caller...
2020/09/28 19:54:24 Service ingress for serviceinvocation-caller is ready...
    service_invocation_test.go:148: externalURL is '192.168.64.5:30824'
=== RUN   TestServiceInvocation/Test_singlehop_for_callee-0
    service_invocation_test.go:160: checking err...
    service_invocation_test.go:164: unmarshalling..{"message":"singlehop is called"}
        
=== RUN   TestServiceInvocation/Test_singlehop_for_callee-1
    service_invocation_test.go:160: checking err...
    service_invocation_test.go:164: unmarshalling..{"message":"singlehop is called"}
        
=== RUN   TestServiceInvocation/Test_multihop
    service_invocation_test.go:160: checking err...
    service_invocation_test.go:164: unmarshalling..{"message":"singlehop is called"}
        
=== RUN   TestServiceInvocation/Test_HTTP_to_HTTP
    service_invocation_test.go:181: url is 'http://192.168.64.5:30824/httptohttptest'
    service_invocation_test.go:186: checking err...
    service_invocation_test.go:190: unmarshalling..{"message":"success"}
        
=== RUN   TestServiceInvocation/Test_HTTP_to_gRPC
    service_invocation_test.go:181: url is 'http://192.168.64.5:30824/httptogrpctest'
    service_invocation_test.go:186: checking err...
    service_invocation_test.go:190: unmarshalling..{"message":"success"}
        
=== RUN   TestServiceInvocation/Test_gRPC_to_HTTP
    service_invocation_test.go:181: url is 'http://192.168.64.5:30824/grpctohttptest'
    service_invocation_test.go:186: checking err...
    service_invocation_test.go:190: unmarshalling..{"message":"success"}
        
=== RUN   TestServiceInvocation/Test_gRPC_to_gRPC
    service_invocation_test.go:181: url is 'http://192.168.64.5:30824/grpctogrpctest'
    service_invocation_test.go:186: checking err...
    service_invocation_test.go:190: unmarshalling..{"message":"success"}
        
--- PASS: TestServiceInvocation (0.12s)
    --- PASS: TestServiceInvocation/Test_singlehop_for_callee-0 (0.02s)
    --- PASS: TestServiceInvocation/Test_singlehop_for_callee-1 (0.02s)
    --- PASS: TestServiceInvocation/Test_multihop (0.02s)
    --- PASS: TestServiceInvocation/Test_HTTP_to_HTTP (0.02s)
    --- PASS: TestServiceInvocation/Test_HTTP_to_gRPC (0.02s)
    --- PASS: TestServiceInvocation/Test_gRPC_to_HTTP (0.02s)
    --- PASS: TestServiceInvocation/Test_gRPC_to_gRPC (0.00s)
=== RUN   TestHeaders
2020/09/28 19:54:24 Waiting until service ingress is ready for serviceinvocation-caller...
2020/09/28 19:54:24 Service ingress for serviceinvocation-caller is ready...
    service_invocation_test.go:212: externalURL is '192.168.64.5:30824'
=== RUN   TestHeaders/http-to-http
    service_invocation_test.go:223: checking err...
    service_invocation_test.go:227: unmarshalling..{"message":"{\"request\":\"{\\\"Accept-Encoding\\\":[\\\"gzip\\\"],\\\"Content-Length\\\":[\\\"38\\\"],\\\"Content-Type\\\":[\\\"application/json\\\"],\\\"Daprtest-Request-1\\\":[\\\"DaprValue1\\\"],\\\"Daprtest-Request-2\\\":[\\\"DaprValue2\\\"],\\\"Forwarded\\\":[\\\"for=172.17.0.13;by=172.17.0.13;host=serviceinvocation-caller-74d568c74c-mgdqx\\\"],\\\"Traceparent\\\":[\\\"00-307f4cfce72ec0269d993c73b1ba9ed3-844f3e47cb273798-00\\\"],\\\"User-Agent\\\":[\\\"Go-http-client/1.1\\\"],\\\"X-Forwarded-For\\\":[\\\"172.17.0.13\\\"],\\\"X-Forwarded-Host\\\":[\\\"serviceinvocation-caller-74d568c74c-mgdqx\\\"]}\",\"response\":\"{\\\"Content-Length\\\":[\\\"463\\\"],\\\"Content-Type\\\":[\\\"application/json; utf-8\\\"],\\\"Daprtest-Response-1\\\":[\\\"DaprTest-Response-Value-1\\\"],\\\"Daprtest-Response-2\\\":[\\\"DaprTest-Response-Value-2\\\"],\\\"Date\\\":[\\\"Tue, 29 Sep 2020 02:54:23 GMT\\\"],\\\"Server\\\":[\\\"fasthttp\\\"],\\\"Traceparent\\\":[\\\"00-307f4cfce72ec0269d993c73b1ba9ed3-60d830c8a39123aa-00\\\"]}\"}"}
        
=== RUN   TestHeaders/grpc-to-grpc
    service_invocation_test.go:265: checking err...
    service_invocation_test.go:269: unmarshalling..{"message":"{\"request\":\"{\\\":authority\\\":[\\\"127.0.0.1:3000\\\"],\\\"content-type\\\":[\\\"application/grpc\\\"],\\\"daprtest-request-1\\\":[\\\"DaprValue1\\\"],\\\"daprtest-request-2\\\":[\\\"DaprValue2\\\"],\\\"forwarded\\\":[\\\"for=172.17.0.13;by=172.17.0.13;host=serviceinvocation-caller-74d568c74c-mgdqx\\\"],\\\"grpc-trace-bin\\\":[\\\"\\\\u0000\\\\u0000\\\\ufffd\\\\n\\\\u001f\\\\ufffd\\\\ufffd9\\\\ufffdK^J\\\\u0018\\\\ufffd\\\\u0014\\\\ufffd\\\\ufffd\\\\u0015\\\\u0001'\\\\u0008\\\\u0018遊h\\\\ufffd\\\\u0002\\\\u0000\\\"],\\\"user-agent\\\":[\\\"grpc-go/1.26.0\\\"],\\\"x-forwarded-for\\\":[\\\"172.17.0.13\\\"],\\\"x-forwarded-host\\\":[\\\"serviceinvocation-caller-74d568c74c-mgdqx\\\"]}\",\"response\":\"{\\\"content-type\\\":[\\\"application/grpc\\\"],\\\"daprtest-response-1\\\":[\\\"DaprTest-Response-Value-1\\\"],\\\"daprtest-response-2\\\":[\\\"DaprTest-Response-Value-2\\\"],\\\"grpc-trace-bin\\\":[\\\"AABL+S81d7NNpqPOkp0ODkc2AQDwZ6oLqQK3AgE=\\\"]}\",\"trailers\":\"{\\\"daprtest-trailer-1\\\":[\\\"DaprTest-Trailer-Value-1\\\"],\\\"daprtest-trailer-2\\\":[\\\"DaprTest-Trailer-Value-2\\\"],\\\"grpc-trace-bin\\\":[\\\"\\\\u0000\\\\u0000\\\\ufffd\\\\n\\\\u001f\\\\ufffd\\\\ufffd9\\\\ufffdK^J\\\\u0018\\\\ufffd\\\\u0014\\\\ufffd\\\\ufffd\\\\u0015\\\\u0001\\\\ufffd}\\\\ufffd\\\\ufffd\\\\\\\"g\\\\ufffd\\\\ufffd\\\\u0002\\\\u0000\\\"]}\"}"}
        
=== RUN   TestHeaders/grpc-to-http
    service_invocation_test.go:312: checking err...
    service_invocation_test.go:316: unmarshalling..{"message":"{\"request\":\"{\\\"Content-Length\\\":[\\\"14\\\"],\\\"Content-Type\\\":[\\\"text/plain; utf-8\\\"],\\\"Dapr-Authority\\\":[\\\"localhost:50001\\\"],\\\"Daprtest-Request-1\\\":[\\\"DaprValue1\\\"],\\\"Daprtest-Request-2\\\":[\\\"DaprValue2\\\"],\\\"Forwarded\\\":[\\\"for=172.17.0.13;by=172.17.0.13;host=serviceinvocation-caller-74d568c74c-mgdqx\\\"],\\\"Traceparent\\\":[\\\"00-793228bf9d6b472297c920ede7236c14-e743033db8e338b8-00\\\"],\\\"User-Agent\\\":[\\\"grpc-go/1.26.0\\\"],\\\"X-Forwarded-For\\\":[\\\"172.17.0.13\\\"],\\\"X-Forwarded-Host\\\":[\\\"serviceinvocation-caller-74d568c74c-mgdqx\\\"]}\",\"response\":\"{\\\"content-length\\\":[\\\"470\\\"],\\\"content-type\\\":[\\\"application/grpc\\\"],\\\"dapr-content-type\\\":[\\\"application/json; utf-8\\\"],\\\"dapr-date\\\":[\\\"Tue, 29 Sep 2020 02:54:24 GMT\\\"],\\\"dapr-http-status\\\":[\\\"200\\\"],\\\"daprtest-response-1\\\":[\\\"DaprTest-Response-Value-1\\\"],\\\"daprtest-response-2\\\":[\\\"DaprTest-Response-Value-2\\\"],\\\"grpc-trace-bin\\\":[\\\"AAB5Mii/nWtHIpfJIO3nI2wUAQ4j/n6hPPGGAgA=\\\"]}\"}"}
        
=== RUN   TestHeaders/http-to-grpc
    service_invocation_test.go:357: checking err...
    service_invocation_test.go:361: unmarshalling..{"message":"{\"request\":\"{\\\":authority\\\":[\\\"127.0.0.1:3000\\\"],\\\"accept-encoding\\\":[\\\"gzip\\\"],\\\"content-length\\\":[\\\"38\\\"],\\\"content-type\\\":[\\\"application/grpc\\\"],\\\"dapr-host\\\":[\\\"localhost:3500\\\"],\\\"daprtest-request-1\\\":[\\\"DaprValue1\\\"],\\\"daprtest-request-2\\\":[\\\"DaprValue2\\\"],\\\"forwarded\\\":[\\\"for=172.17.0.13;by=172.17.0.13;host=serviceinvocation-caller-74d568c74c-mgdqx\\\"],\\\"grpc-trace-bin\\\":[\\\"\\\\u0000\\\\u0000Ӓ\\\\ufffd7N\\\\ufffdt\\\\r\\\\ufffd\\\\ufffdhOS\\\\ufffdj\\\\ufffd\\\\u0001hQ\\\\ufffde\\\\ufffd\\\\ufffd\\\\ufffd\\\\ufffd\\\\u0002\\\\u0000\\\"],\\\"user-agent\\\":[\\\"grpc-go/1.26.0\\\"],\\\"x-forwarded-for\\\":[\\\"172.17.0.13\\\"],\\\"x-forwarded-host\\\":[\\\"serviceinvocation-caller-74d568c74c-mgdqx\\\"]}\",\"response\":\"{\\\"Content-Length\\\":[\\\"580\\\"],\\\"Content-Type\\\":[\\\"application/json\\\"],\\\"Daprtest-Response-1\\\":[\\\"DaprTest-Response-Value-1\\\"],\\\"Daprtest-Response-2\\\":[\\\"DaprTest-Response-Value-2\\\"],\\\"Date\\\":[\\\"Tue, 29 Sep 2020 02:54:23 GMT\\\"],\\\"Server\\\":[\\\"fasthttp\\\"],\\\"Traceparent\\\":[\\\"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01\\\"]}\"}"}
        
=== RUN   TestHeaders/http-to-http-tracing
    service_invocation_test.go:422: checking err...
    service_invocation_test.go:426: unmarshalling..{"message":"{\"request\":\"{\\\"Accept-Encoding\\\":[\\\"gzip\\\"],\\\"Content-Length\\\":[\\\"38\\\"],\\\"Content-Type\\\":[\\\"application/json\\\"],\\\"Daprtest-Request-1\\\":[\\\"DaprValue1\\\"],\\\"Daprtest-Request-2\\\":[\\\"DaprValue2\\\"],\\\"Daprtest-Traceid\\\":[\\\"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01\\\"],\\\"Forwarded\\\":[\\\"for=172.17.0.13;by=172.17.0.13;host=serviceinvocation-caller-74d568c74c-mgdqx\\\"],\\\"Traceparent\\\":[\\\"00-24f02fae8723d50950e2e4d43feb5af4-4a38c832a59f3ad8-00\\\"],\\\"User-Agent\\\":[\\\"Go-http-client/1.1\\\"],\\\"X-Forwarded-For\\\":[\\\"172.17.0.13\\\"],\\\"X-Forwarded-Host\\\":[\\\"serviceinvocation-caller-74d568c74c-mgdqx\\\"]}\",\"response\":\"{\\\"Content-Length\\\":[\\\"542\\\"],\\\"Content-Type\\\":[\\\"application/json; utf-8\\\"],\\\"Daprtest-Response-1\\\":[\\\"DaprTest-Response-Value-1\\\"],\\\"Daprtest-Response-2\\\":[\\\"DaprTest-Response-Value-2\\\"],\\\"Date\\\":[\\\"Tue, 29 Sep 2020 02:54:23 GMT\\\"],\\\"Server\\\":[\\\"fasthttp\\\"],\\\"Traceparent\\\":[\\\"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01\\\"]}\"}"}
        
=== RUN   TestHeaders/grpc-to-grpc-tracing
    service_invocation_test.go:455: checking err...
    service_invocation_test.go:459: unmarshalling..{"message":"{\"request\":\"{\\\":authority\\\":[\\\"127.0.0.1:3000\\\"],\\\"content-type\\\":[\\\"application/grpc\\\"],\\\"daprtest-request-1\\\":[\\\"DaprValue1\\\"],\\\"daprtest-request-2\\\":[\\\"DaprValue2\\\"],\\\"daprtest-traceid\\\":[\\\"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01\\\"],\\\"forwarded\\\":[\\\"for=172.17.0.13;by=172.17.0.13;host=serviceinvocation-caller-74d568c74c-mgdqx\\\"],\\\"grpc-trace-bin\\\":[\\\"\\\\u0000\\\\u0000\\\\ufffdq=\\\\ufffd\\\\ufffd\\\\ufffd2Ma\\\\ufffdG\\\\u000b\\\\ufffdń\\\\ufffd\\\\u0001\\\\ufffd\\\\ufffdx\\\\ufffd\\\\ufffd\\\\ufffd\\\\u0004\\\\u001b\\\\u0002\\\\u0000\\\"],\\\"user-agent\\\":[\\\"grpc-go/1.26.0\\\"],\\\"x-forwarded-for\\\":[\\\"172.17.0.13\\\"],\\\"x-forwarded-host\\\":[\\\"serviceinvocation-caller-74d568c74c-mgdqx\\\"]}\",\"response\":\"{\\\"content-type\\\":[\\\"application/grpc\\\"],\\\"daprtest-response-1\\\":[\\\"DaprTest-Response-Value-1\\\"],\\\"daprtest-response-2\\\":[\\\"DaprTest-Response-Value-2\\\"],\\\"grpc-trace-bin\\\":[\\\"AABL+S81d7NNpqPOkp0ODkc2AQDwZ6oLqQK3AgE=\\\"]}\",\"trailers\":\"{\\\"daprtest-trailer-1\\\":[\\\"DaprTest-Trailer-Value-1\\\"],\\\"daprtest-trailer-2\\\":[\\\"DaprTest-Trailer-Value-2\\\"],\\\"grpc-trace-bin\\\":[\\\"\\\\u0000\\\\u0000\\\\ufffdq=\\\\ufffd\\\\ufffd\\\\ufffd2Ma\\\\ufffdG\\\\u000b\\\\ufffdń\\\\ufffd\\\\u0001\\\\u0013\\\\u00132\\\\u0011\\\\u001e\\\\ufffd%R\\\\u0002\\\\u0000\\\"]}\"}"}
        
    service_invocation_test.go:480: received response grpc header..AABL+S81d7NNpqPOkp0ODkc2AQDwZ6oLqQK3AgE=
=== RUN   TestHeaders/http-to-grpc-tracing
    service_invocation_test.go:500: checking err...
    service_invocation_test.go:504: unmarshalling..{"message":"{\"request\":\"{\\\":authority\\\":[\\\"127.0.0.1:3000\\\"],\\\"accept-encoding\\\":[\\\"gzip\\\"],\\\"content-length\\\":[\\\"38\\\"],\\\"content-type\\\":[\\\"application/grpc\\\"],\\\"dapr-host\\\":[\\\"localhost:3500\\\"],\\\"daprtest-request-1\\\":[\\\"DaprValue1\\\"],\\\"daprtest-request-2\\\":[\\\"DaprValue2\\\"],\\\"daprtest-traceid\\\":[\\\"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01\\\"],\\\"forwarded\\\":[\\\"for=172.17.0.13;by=172.17.0.13;host=serviceinvocation-caller-74d568c74c-mgdqx\\\"],\\\"grpc-trace-bin\\\":[\\\"\\\\u0000\\\\u0000\\\\ufffd\\\\ufffd\\\\ufffdg\\\\ufffd\\\\u001blն\\\\ufffd\\\\u001c\\\\u000e\\\\ufffd\\\\u001e\\\\ufffdB\\\\u0001\\\\ufffd\\\\ufffd(_\\\\ufffd\\\\u0026SU\\\\u0002\\\\u0000\\\"],\\\"user-agent\\\":[\\\"grpc-go/1.26.0\\\"],\\\"x-forwarded-for\\\":[\\\"172.17.0.13\\\"],\\\"x-forwarded-host\\\":[\\\"serviceinvocation-caller-74d568c74c-mgdqx\\\"]}\",\"response\":\"{\\\"Content-Length\\\":[\\\"678\\\"],\\\"Content-Type\\\":[\\\"application/json\\\"],\\\"Daprtest-Response-1\\\":[\\\"DaprTest-Response-Value-1\\\"],\\\"Daprtest-Response-2\\\":[\\\"DaprTest-Response-Value-2\\\"],\\\"Date\\\":[\\\"Tue, 29 Sep 2020 02:54:23 GMT\\\"],\\\"Server\\\":[\\\"fasthttp\\\"],\\\"Traceparent\\\":[\\\"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01\\\"]}\"}"}
        
=== RUN   TestHeaders/grpc-to-http-tracing
    service_invocation_test.go:533: checking err...
    service_invocation_test.go:537: unmarshalling..{"message":"{\"request\":\"{\\\"Content-Length\\\":[\\\"14\\\"],\\\"Content-Type\\\":[\\\"text/plain; utf-8\\\"],\\\"Dapr-Authority\\\":[\\\"localhost:50001\\\"],\\\"Daprtest-Request-1\\\":[\\\"DaprValue1\\\"],\\\"Daprtest-Request-2\\\":[\\\"DaprValue2\\\"],\\\"Daprtest-Traceid\\\":[\\\"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01\\\"],\\\"Forwarded\\\":[\\\"for=172.17.0.13;by=172.17.0.13;host=serviceinvocation-caller-74d568c74c-mgdqx\\\"],\\\"Traceparent\\\":[\\\"00-cbde493834fcd1e353618afcc1ec989d-ad2c8d28925b3cf8-00\\\"],\\\"User-Agent\\\":[\\\"grpc-go/1.26.0\\\"],\\\"X-Forwarded-For\\\":[\\\"172.17.0.13\\\"],\\\"X-Forwarded-Host\\\":[\\\"serviceinvocation-caller-74d568c74c-mgdqx\\\"]}\",\"response\":\"{\\\"content-length\\\":[\\\"549\\\"],\\\"content-type\\\":[\\\"application/grpc\\\"],\\\"dapr-content-type\\\":[\\\"application/json; utf-8\\\"],\\\"dapr-date\\\":[\\\"Tue, 29 Sep 2020 02:54:24 GMT\\\"],\\\"dapr-http-status\\\":[\\\"200\\\"],\\\"daprtest-response-1\\\":[\\\"DaprTest-Response-Value-1\\\"],\\\"daprtest-response-2\\\":[\\\"DaprTest-Response-Value-2\\\"],\\\"grpc-trace-bin\\\":[\\\"AABL+S81d7NNpqPOkp0ODkc2AQDwZ6oLqQK3AgE=\\\"]}\"}"}
        
    service_invocation_test.go:556: received response grpc header..AABL+S81d7NNpqPOkp0ODkc2AQDwZ6oLqQK3AgE=
--- PASS: TestHeaders (0.06s)
    --- PASS: TestHeaders/http-to-http (0.00s)
    --- PASS: TestHeaders/grpc-to-grpc (0.01s)
    --- PASS: TestHeaders/grpc-to-http (0.00s)
    --- PASS: TestHeaders/http-to-grpc (0.01s)
    --- PASS: TestHeaders/http-to-http-tracing (0.00s)
    --- PASS: TestHeaders/grpc-to-grpc-tracing (0.01s)
    --- PASS: TestHeaders/http-to-grpc-tracing (0.00s)
    --- PASS: TestHeaders/grpc-to-http-tracing (0.02s)
PASS
2020/09/28 19:54:24 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/service_invocation	38.993s
2020/09/28 19:54:26 Running setup...
2020/09/28 19:54:27 Installing test apps...
2020/09/28 19:54:27 Adding app {stateapp 0  true e2e-stateapp:dev-linux-amd64 artursouza 1 true   4.0 0.5 512Mi 250Mi}
2020/09/28 19:54:39 Running tests...
=== RUN   TestStateApp
2020/09/28 19:54:39 Waiting until service ingress is ready for stateapp...
2020/09/28 19:54:39 Service ingress for stateapp is ready...
=== RUN   TestStateApp/Test_get_getbulk_save_delete_with_empty_request_response_for_single_app_and_single_hop
=== RUN   TestStateApp/Test_save_get_and_delete_a_single_item_for_single_app_and_single_hop
=== RUN   TestStateApp/Test_save_get_and_delete_on_multiple_items_for_single_app_and_single_hop
--- PASS: TestStateApp (1.17s)
    --- PASS: TestStateApp/Test_get_getbulk_save_delete_with_empty_request_response_for_single_app_and_single_hop (0.00s)
    --- PASS: TestStateApp/Test_save_get_and_delete_a_single_item_for_single_app_and_single_hop (0.01s)
    --- PASS: TestStateApp/Test_save_get_and_delete_on_multiple_items_for_single_app_and_single_hop (0.02s)
=== RUN   TestStateTransactionApps
2020/09/28 19:54:40 Waiting until service ingress is ready for stateapp...
2020/09/28 19:54:40 Service ingress for stateapp is ready...
=== RUN   TestStateTransactionApps/Test_State_Transactions_using_HTTP_protocol
=== RUN   TestStateTransactionApps/Test_State_Transactions_using_GRPC_protocol
--- PASS: TestStateTransactionApps (0.01s)
    --- PASS: TestStateTransactionApps/Test_State_Transactions_using_HTTP_protocol (0.00s)
    --- PASS: TestStateTransactionApps/Test_State_Transactions_using_GRPC_protocol (0.01s)
PASS
2020/09/28 19:54:40 Running teardown...
ok  	github.com/dapr/dapr/tests/e2e/stateapp	13.897s
?   	github.com/dapr/dapr/tests/e2e/utils	[no test files]
```

**Comment by artursouza** [comment]
Make sure you start minikube with `minikube start --cpus=4 --driver=hyperkit `

**Comment by artursouza** [comment]
There is a PR to run E2E tests on KinD. It takes about 20 minutes: https://github.com/dapr/dapr/actions/runs/278849209
I created a draft PR to run E2E tests on Minikube, it is taking about 2h and the run gets cancelled: https://github.com/dapr/dapr/runs/1180749174?check_suite_focus=true

Running E2E locally with Minikube on my Macbook Pro was also very slow. My proposal is for KinD to be the recommended way to run E2E tests locally instead of Minikube.

**Comment by artursouza** [comment]
Team has agreed to use KinD instead of Minikube to run E2E tests locally. Closing this in favor of https://github.com/dapr/dapr/issues/2157
